lib/spack/spack/modules/lmod.py

```diff
@@ -283,16 +283,18 @@ def token_to_path(self, name, value):
 
         # If we are dealing with a core compiler, return 'Core'
         core_compilers = self.conf.core_compilers
```

Comment:
Can you make sure `core_compilers` is a list of CompilerSpec?

Reply:
Does that give us more guarantee when used as an argument to satisfies?